test/wrapper_test.rb

```diff
@@ -3225,6 +3225,67 @@ def test_cannot_simplify_service_setup_duration
     }
   end
 
+  def test_prioritize_first_available_trips_and_vehicles
+    # Solve WITH simplification and note the cost and check if trips are not skipped
+    vrp = TestHelper.load_vrp(self, fixture_file: 'vrp_multi_trips_which_uses_trip2_before_trip1')
+
+    soln_with_simplification = OptimizerWrapper.wrapper_vrp('demo', { services: { vrp: [:ortools] }}, vrp, nil)
+
+    there_is_no_skipped_trip_when_simplification_is_on = vrp.relations.none?{ |relation|
+      skipped_a_trip = false
+      relation.linked_vehicle_ids.any?{ |vid|
+        route = soln_with_simplification[:routes].find{ |r| r[:vehicle_id] == vid }
+
+        # if an earlier trip is skipped check if a later trip is active
+        if skipped_a_trip
+          route[:end_time] > route[:start_time]
+        else
+          skipped_a_trip = (route[:end_time] <= route[:start_time])
+          nil
+        end
+      }
+    }
+    assert there_is_no_skipped_trip_when_simplification_is_on, 'Should prevent skipped trips for this instance'
+
+    # Solve WITHOUT simplification (when prioritize_first_available_trips_and_vehicles is off)
+    # verify the cost is the same or better with the simplification and that the trip2 is used before trip1
+    simplifier_called = false
+    soln_without_simplification = Wrappers::Wrapper.stub_any_instance(:prioritize_first_available_trips_and_vehicles,
+                                                                      proc{
+                                                                        simplifier_called = true
+                                                                        nil
+                                                                      }) do
+      vrp = TestHelper.load_vrp(self, fixture_file: 'vrp_multi_trips_which_uses_trip2_before_trip1')
+      OptimizerWrapper.wrapper_vrp('ortools', { services: { vrp: [:ortools] }}, vrp, nil)
+    end
+    assert simplifier_called, 'prioritize_first_available_trips_and_vehicles should have been called'
```

Comment:
It is called `simplifier_called` because the intention of that assert is really to make sure that the simplifier function is called. 

Even though the variable assignment happens inside the empty/stubbed version of the function, the assert/variable is still checking if the simplifier (`prioritize_first_available_trips_and_vehicles`) is called or not. This covers the main case (without the stub) as well because the only difference is the function itself being stubbed.

If you prefer, I can do `function_called` or `simplification_function_called`?

Reply:
_function_called_ is good for me. It removes the link with simplification : we do want to call the function, but we do not want to call the simplifcation .

Reply:
Technically speaking, we DO want the code to call the simplifier function named `prioritize_first_available_trips_and_vehicles` and we DO want to verify if this specific simplifier is called or not; ergo the var name `simplifier_called` (the function being stubbed to be empty at that moment doesn't affect this function call verification logic). 

A routine being called is not the same thing as it being active or something being applied. I would agree :100: if the variable was named _simplification_applied_ or _simplification_active_ etc. We can agree to disagree. It is a bit potayto, potahto anyways.